torch/distributions/multinomial.py

```diff
@@ -103,6 +105,20 @@ def sample(self, sample_shape=torch.Size()):
         counts.scatter_add_(-1, samples, torch.ones_like(samples))
         return counts.type_as(self.probs)
 
+    def entropy(self):
+        n = torch.tensor(self.total_count)
+
+        cat_entropy = self._categorical.entropy()
+        log_factorial_n = torch.lgamma(n + 1) * torch.ones_like(cat_entropy)
+        term1 = -log_factorial_n + n * cat_entropy
+
+        support = self._binomial.enumerate_support(expand=False)[1:]
+        binomial_probs = torch.exp(self._binomial.log_prob(support))
+        weights = torch.lgamma(support + 1)
+        term2 = (binomial_probs * weights).sum([0, -1])
```

Comment:
do you mean to `.sum([-2, -1])`? I don't fully follow the logic, but `.sum(0)` will break batching.

Reply:
As same as the `Binomial.entropy()` function, I tried to take a summation over `binomial support dimension = 0`, and `multinomial probability parameter dimension = -1`.

Reply:
Oh thanks for explaining, that now makes sense!